fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/AgilityContext.java

```diff
@@ -499,6 +513,11 @@ public void abortAndClose() {
         public boolean isClosed() {
             return closed;
         }
+
+        @Override
+        public void setPreCommit(final Function<FDBRecordContext, CompletableFuture<Void>> preCommit) {
+            callerContext.addCommitCheck(() -> preCommit.apply(callerContext));
```

Comment:
Letting the non-agile implement it this way is a nice abstraction . I can't think of any issue expected in our code, but also not much gain - as the user context will commit after directory close. 
@ScottDugas, do you see any issue with it?